sdk/java/pulumi/src/test/java/com/pulumi/resources/internal/StackTest.java

```diff
@@ -109,19 +106,27 @@ void testStackWithNullOutputsThrows() {
     @Test
     void testImmediateStackReference() {
         var monitorMocks = new RecurrentStackMocks();
-        var mock = DeploymentMockBuilder.builder()
+        var test = PulumiTestInternal.builder()
+                .mocks(monitorMocks)
                 .deploymentFactory(state -> Mockito.spy(new DeploymentImpl(state)))
-                .setMocks(monitorMocks)
                 .build();
-        monitorMocks.setMock(mock);
 
-        var result = mock.runTestAsync(ctx -> {
-            assertThat(((MockMonitor) mock.monitor).resources)
+        monitorMocks.setStack(() -> {
```

Comment:
Lots of places use this pattern of finding resources by type. I think we might use a helper method in the testing framework.